javascripts/tagmanagerHelper.js

```diff
@@ -173,9 +173,47 @@
         var params = {method: 'TagManager.disablePreviewMode', idContainer: idContainer};
         piwikHelper.modalConfirm('<h2>' + _pk_translate('TagManager_DisablingPreviewPleaseWait') + '</h2>', {});
         piwikApi.fetch(params).then(function () {
+            tagManagerHelper.updateDebugSiteFlag(document.getElementById('previewDebugUrl').value, idContainer, -1);
             window.location.reload();
         });
     };
+    tagManagerHelper.changeDebugUrl = function (idContainer, oldUrl) {
+        var newUrl = document.getElementById('previewDebugUrl').value;
+        var id = 'TagManager_changeDebugSiteUrl';
+        var context = 'warning'; // or 'warning' or 'error' or 'success'
+        var UI = require('piwik/UI');
+        var notification = new UI.Notification();
+        if (!newUrl) {
+            return notification.show(_pk_translate('TagManager_DebugUrlNoUrlErrorMessage'), {context: context, id: id, title: ''});
+        } else if (newUrl === oldUrl) {
+            return tagManagerHelper.updateDebugSiteFlag(newUrl, idContainer, 1);;
+        }
+        var ajaxRequest = new ajaxHelper();
+        ajaxRequest.addParams({
+            module: 'API',
+            method: 'TagManager.changeDebugUrl',
+            idContainer: idContainer,
+            url: newUrl
+        }, 'get');
+        ajaxRequest.withTokenInUrl();
+        ajaxRequest.setCallback(
+            function (response) {
+                tagManagerHelper.updateDebugSiteFlag(oldUrl, idContainer, -1);
+                tagManagerHelper.updateDebugSiteFlag(newUrl, idContainer, 1);
+                window.location.reload();
+            }
+        );
+        ajaxRequest.setFormat('html');
+        piwikHelper.modalConfirm('<h2>' + _pk_translate('TagManager_UpdatingDebugSiteUrlPleaseWait') + '</h2>', {});
+        ajaxRequest.send();
+    };
+    tagManagerHelper.updateDebugSiteFlag = function (url, idContainer, debugFlag) {
+        if (!url || !idContainer || !debugFlag) {
+            return;
+        }
+        window.open(url + (url.indexOf('?') == -1 ? '?' : '&') + 'mtmPreviewMode=' + idContainer + '&mtmSetDebugFlag=' + debugFlag, '_blank', 'noopener');
```

Comment:
can we also set `noreferrer` here next to `noopener` (technically noreferrer will also set noopener but might not hurt to mention both)?

Reply:
also here we will also url encode all the `idContainer` and the `debugFlag` just to be safe.

Reply:
set to noreferrer and added encodeURIComponent